src/test_util/mod.rs

```diff
@@ -50,6 +50,14 @@ macro_rules! assert_downcast_matches {
     }};
 }
 
+pub fn test_generate_config<T>()
```

Comment:
This is great!

Reply:
Could we also have a version that allows passing in an asserting the expected value?

Reply:
Yes we can! However, I don't want to use that at the moment, having to write a Toml::Table to compare seems like a tedious task..

Reply:
It doesn't look like it's required in this PR (as it won't really be using it), but for any custom implementation, it would be required to effectively test the outcomes.

Maybe it could be done by comparing the values that are serialized to string, rather than via raw `Toml::Table`s?